HPCC-19479 Create WsELK for ELK Stack integration

rpastrana opened the pull request:
- Declares ECLWatch configuration for ELK Stack integration
- Provides integration configuration

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

<!-- Thank you for submitting a pull request to the HPCC project

 PLEASE READ the following before proceeding.

 This project only accepts pull requests related to open JIRA issues.
 If suggesting a new feature or change, please discuss it in a JIRA issue first.
 If fixing a bug, there should be an issue describing it with steps to reproduce.
 The title line of the pull request (and of each commit within it) should refer to the
 associated issue using the format:

 HPCC-nnnnn Short description of issue

 This will allow the Jira ticket to be automatically updated to refer to this pull request,
 and will ensure that the automatically-generated changelog is properly formatted.
 Where a pull request contains a single commit the pull request title will be set automatically,
 assuming that the commit has followed the proper guidelines.

 Please go over all the following points, and put an `x` in all the boxes that apply. You may find
 it easier to press the 'Create' button first then click on the checkboxes to edit the comment.
-->

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [x] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Testing:
<!-- Please describe how this change has been tested.-->

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

rpastrana committed:
HPCC-19479 Create WsELK for ELK Stack integration

- Declares ECLWatch configuration for ELK Stack integration
- Provides integration configuration

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana commented:
@miguelvazq please confirm this service provides the information you need to embed Kibana dashboards
You can expect the information to be returned in this format:
{
  "GetConfigDetailsResponse": {
    "IntegrateKibana": true,
    "KibanaAddress": "String",
    "KibanaPort": "String",
    "KibanaEntryPointURI": "String",
    "ReportElasticSearchHealth": true,
    "ElasticSearchAdresses": "String",
    "ElasticSearchPort": "String",
    "ReportLogStashHealth": true,
    "LogStashAddress": "String",
    "LogStashPort": "String"
  }
}

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: ce6618a8a58600047b3cb6cdeb188792d95fd891
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |88 |88 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |776 |776 |0 |
|test (thor) |706 |706 |0 |
|test (roxie) |819 |819 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|   14 sec (00:00:14) |  183 sec (00:03:03) |   56 sec (00:00:56) |    6 sec (00:00:06) |   21 sec (00:00:21) | 1168 sec (00:19:28) |   17 sec (00:00:17) | 1465 sec (00:24:25) |

miguelvazq commented:
This could grow as we keep think of coming up with feature sets. But, at first glace and our discussion looks good. Only concern would be a security issue sending requests via ajax to IP's that are locked down.

rpastrana commented:
@wangkx please review.

wangkx commented on esp/scm/ws_elk.ecm:
```diff
@@ -0,0 +1,27 @@
+
+ESPservice [auth_feature("ESDLConfigAccess:ACCESS"), version("1.0"), default_client_version("1.0"), exceptions_inline("xslt/exceptions.xslt")] ws_elk
+{
+    ESPmethod getConfigDetails(GetConfigDetailsRequest, GetConfigDetailsResponse);
```
I prefer that the method name starts with Capital ('G') as other ESP services.

rpastrana replied:
Indeed! Not sure how that slipped through

wangkx commented on initfiles/componentfiles/configxml/espsmcservice.xsd.in:
```diff
@@ -49,10 +49,10 @@
+              <xs:appinfo>
+              <!--DOCS_Autobuild_Code-->
+                <docid>SMC-T01</docid>
+              </xs:appinfo>
```
wrong indent in at least 1 line here

wangkx commented on initfiles/componentfiles/configxml/espsmcservice.xsd.in:
```diff
@@ -153,15 +153,99 @@
     <xs:attributeGroup name="Monitoring">
          <xs:annotation>
-	    <xs:appinfo>
+        <xs:appinfo>
```
wrong indent

wangkx reviewed (commented):
@rpastrana  just a few minor comments.

rpastrana commented on initfiles/componentfiles/configxml/espsmcservice.xsd.in:
```diff
@@ -49,10 +49,10 @@
-	          <!--DOCS_Autobuild_Code-->
-          	    <docid>SMC-T01</docid>
-	          </xs:appinfo>
+              <xs:appinfo>
```
@g-pan @JamesDeFabia please comment on what this element should be and if there are any missing docid elements, etc.

rpastrana committed:
HPCC-19479 Create WsELK for ELK Stack integration

- Code Review - format cleanup
- Capitalizes method name
- Adds default port for elastic search

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexis.com>

rpastrana commented:
@wangkx please review.

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.2.1511 (Linux 3.10.0-327.28.3.el7.x86_64)
Sha: 77ed2964428032b61a422d20cafcd846d3f7b5b6
Build: success
Install hpccsystems-platform-community_6.5.0-trunk0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | 
|---|---|---|---|---|---|
|unittest |88 |88 |0 |0 |0 |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |



Regression test result:

| phase | total | pass | fail | 
|---|---|---|---|
|setup (hthor) |11 |11 |0 |
|setup (thor) |11 |11 |0 |
|setup (roxie) |11 |11 |0 |
|test (hthor) |776 |776 |0 |
|test (thor) |707 |707 |0 |
|test (roxie) |820 |820 |0 |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    6 sec (00:00:06) |  182 sec (00:03:02) |   55 sec (00:00:55) |    6 sec (00:00:06) |   22 sec (00:00:22) | 1169 sec (00:19:29) |   16 sec (00:00:16) | 1456 sec (00:24:16) |

wangkx reviewed (approved):
@rpastrana looks good.
